### PR TITLE
[SW-615] fix: do not show all WAE transactions as DEX

### PR DESCRIPTION
**Pull request by CedrikNikita**

### CODE REVIEW

File: src/composables/transactionTx.ts [hunk 1]
```diff
@@ -89,6 +90,11 @@ export function useTransactionTx({
 
     return !!(
       innerTx.value?.contractId
+      && innerTx.value?.function
+      && (
+        Object.values(flatten(Object.values(FUNCTION_TYPE_DEX)))
+          .includes(innerTx.value?.function as TxFunctionRaw)
+      )
```
Comment (pstachel) [inline src/composables/transactionTx.ts:97]: I don't see the need for the first `Object.values()` call. Also we could reduce our dependency on external libraries and use plain js (but that's just a suggestion).
```suggestion
      && (
        Object.values(FUNCTION_TYPE_DEX).flat()
          .includes(innerTx.value?.function as TxFunctionRaw)
      )
```

File: src/composables/transactionTx.ts [hunk 2]
```diff
@@ -89,6 +89,11 @@ export function useTransactionTx({
 
     return !!(
       innerTx.value?.contractId
+      && innerTx.value?.function
+      && (
+        Object.values(FUNCTION_TYPE_DEX).flat()
```
Comment (peronczyk) [inline src/composables/transactionTx.ts:94]: I suggest to use `Object.values(TX_FUNCTIONS)` here.

File: src/popup/components/Modals/ConfirmTransactionSign.vue [hunk 3]
```diff
@@ -222,13 +219,10 @@ export default defineComponent({
       return 'total';
     });
 
-    const totalAmount = computed(() => getTxAmountTotal.value(
-      txWrapped.value,
-      getTxDirection.value(props.transaction),
-    ));
+    const totalAmount = computed(() => getTxAmountTotal.value(txWrapped.value, direction.value));
```
Comment (peronczyk) [inline src/popup/components/Modals/ConfirmTransactionSign.vue:222]: This change makes the code less readable in my opinion. Less lines != better code :)

File: src/popup/components/Modals/ConfirmTransactionSign.vue [hunk 4]
```diff
@@ -222,13 +219,10 @@ export default defineComponent({
...
 
     const singleToken = computed((): ITokenResolved => ({
-      isReceived: getTxDirection.value(props.transaction) === TX_FUNCTIONS.received,
+      isReceived: direction.value === TX_FUNCTIONS.received,
```
Comment (peronczyk) [inline src/popup/components/Modals/ConfirmTransactionSign.vue:225]: Side note: isn't this a mistake that the direction values are stored in the `TX_FUNCTION` collection?

File: src/composables/transactionTx.ts (hunk 2 above)
Comment (CedrikNikita) [reply src/composables/transactionTx.ts:94]: We are searching only for DEXfunctions, but `TX_FUNCTIONS` contain all function not only DEX.

**Review by peronczyk — APPROVED** [review]

File: src/popup/components/Modals/ConfirmTransactionSign.vue (hunk 4 above)
Comment (CedrikNikita) [reply src/popup/components/Modals/ConfirmTransactionSign.vue:225]: Looks like a mistake to me.

File: src/composables/transactionTx.ts (hunk 2 above)
Comment (peronczyk) [reply src/composables/transactionTx.ts:94]: Uhhh, OK. Thanks for clarification.

**Review by pstachel — APPROVED** [review]

**Pull request merged**